tests/workflows/test_ubicloud_runner_placement.py

```
@@ -0,0 +1,311 @@
+    provider must not exempt its job from the placement checks.
+    """
+    labels = _mapping_labels(selector)
+    if labels is None:
+        return None
+    group = selector.get("group")
+    if group is None:
+        return labels
+    return [*labels, group] if isinstance(group, str) else None
+
+
+def _labels_from_selector(selector: Any) -> list[str] | None:
```

Ruff (0.16.3)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `selector`

(ANN401)